src/components/body/BodyController.js

```diff
@@ -12,74 +12,112 @@ export class BodyController{
   constructor($scope, $timeout){
     this.$scope = $scope;
     this.tempRows = [];
+    this.watchListeners = [];
 
+    this.setTreeAndGroupColumns();
+    this.setConditionalWatches();
+
+    $scope.$watch('body.options.columns', (newVal, oldVal) => {
+      if (newVal) {
+        const origTreeColumn = this.treeColumn,
+          origGroupColumn = this.groupColumn;
+
+        this.setTreeAndGroupColumns();
+
+        this.setConditionalWatches();
```

Comment:
this might be more accurate be on a watch for options.scrollbarV && options.paging.externalPaging